### PR TITLE
decode messages to wire message

### PR DESCRIPTION
**Pull request by kishansagathiya**
collation protocol and validation protocol uses messages wrapped as wiremessage. So we had to adjust our decoder and handler functions according to that.

Apart from this I have also added message type to wire message struct. Since wire message could be collation message or validation message, I have chose to manually set this as a value instead of returning a hardcoded value.

## Tests

```sh
go test -tags integration github.com/ChainSafe/gossamer
```

## Issues
#4108

### CODE REVIEW

**Review by edwardmack — APPROVED** [review]
Reviewed together in design meeting.  Looks good.

**Review by axaysagathiya — APPROVED** [review]

File: dot/parachain/network-bridge/wire_message.go [hunk 1]
```diff
@@ -67,9 +68,13 @@ func (mvdt WireMessage) ValueAt(index uint) (value any, err error) {
 	return nil, scale.ErrUnknownVaryingDataTypeValue
 }
 
+func (w *WireMessage) SetType(messageType network.MessageType) {
+	w.messageType = messageType
+}
+
 func (w WireMessage) Type() network.MessageType {
 	// TODO: create a wire message type and return that #4108
```
Comment (EclesioMeloJunior) [inline dot/parachain/network-bridge/wire_message.go]: I think this TODO is resolved now, right?

**Review by EclesioMeloJunior — APPROVED** [review]
LGTM! just a small comment

File: dot/parachain/network-bridge/wire_message.go (hunk 1 above)
Comment (kishansagathiya) [reply dot/parachain/network-bridge/wire_message.go]:
```suggestion
```

Comment (kishansagathiya) [reply dot/parachain/network-bridge/wire_message.go]: yes, it is

**Pull request merged**